src/mongo/db/audit/audit.cpp

```diff
@@ -214,6 +227,101 @@ namespace audit {
         SimpleMutex _mutex;
     };    
 
+    // Writes audit events to a json file
+    class JSONAuditLog: public FileAuditLog {
+
+        class Adapter: public AuditLogFormatAdapter {
+            const std::string str;
+
+        public:
+            Adapter(const BSONObj &obj)
+                : str(mongoutils::str::stream() << obj.jsonString() << "\n") {}
```

Comment:
It's cheaper to keep `mongoutils::str::stream` as a member instead of `std::string`, and extract data and size on-the-fly from this `stream` object when requested.